sbin/shipper_utils/update_and_install_workers.py

```diff
@@ -52,6 +52,34 @@ def update_docker_images(user, host, worker, autograding_workers, autograding_co
     if host == "localhost":
         get_sysinfo.print_distribution()
         client = docker.from_env()
+        try:
+            # List all images
+            image_set = {
+                image_name
+                for image in client.images.list()
+                for image_name in image.attrs["RepoTags"]
+            }
+
+            images_to_remove = set.difference(image_set, images_to_update)
+
+            # Prevent removal of system docker containers
+            with open("/usr/local/submitty/GIT_CHECKOUT/Submitty/.setup/data/system_docker_containers.json") as json_file:
+                system_docker_containers = json.load(json_file)
+
+            images_to_remove = set.difference(images_to_remove, set(system_docker_containers))
+
+            # Remove images
+            for imageRemoved in images_to_remove:
+                try:
+                    image_id = client.images.get(imageRemoved).id
+                    print("Removed image " + imageRemoved)
+                except docker.errors.ImageNotFound as e:
+                    print("Couldn't find image ", e)
```

Comment:
This message could easily be lost in the logs without someone realizing that an error occurred.  It would be better to explicitly add the text "ERROR:" to this string, like the other error messages in this file.  Ideally we should also print errors to stderr, but that's a broader issue outside the scope of this PR.